packages/toolboxes/evm/src/toolbox/BaseEVMToolbox.ts

```diff
@@ -439,10 +445,14 @@ const sendTransaction = async (
       ...feeData,
     };
 
-    const txHex = await signer.signTransaction(txObject);
-    const response = await provider.broadcastTransaction(txHex);
-
-    return typeof response?.hash === 'string' ? response.hash : response;
+    try {
+      const response = await signer.sendTransaction(txObject);
+      return typeof response?.hash === 'string' ? response.hash : response;
+    } catch (error) {
+      const txHex = await signer.signTransaction(txObject);
+      const response = await provider.broadcastTransaction(txHex);
+      return typeof response?.hash === 'string' ? response.hash : response;
+    }
```

Comment:
Isn't there possibility that sth inner `sendTransaction` fails but still broadcasts tx and then we will end up with double broadcast? 🤔